src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -132,9 +136,15 @@ public List<EventBridgeResult<EventBridgeEventId>> putItems(List<SinkRecord> rec
       return mappingResult.getErrorsAsResult();
     }
 
```

Comment:
nit: please add a comment above mappingResult to abstract this hardcoded pipeline into a more generic implementation.

Reply:
Will be replaced if WIP is finished by constructor parameter.

Reply:
Not quite sure if this is what @embano1 is asking for here but in this version it seems the default path is always dealing with offloaded messages (offLoadingResult), while I would argue this is a rather "exceptional" case for most users. I would suggest we try to be either more explicit by filtering for messages that have to be offloaded and handle them separately or be more generic without making offloading explicit e.g. by hiding the fact that things are getting offloaded and the writer simply deals with mappingResults.

Reply:
@maschnetwork sorry, we haven't published our design/QnA yet that @agebhar1 and I discussed as the foundation for the proposal. This will give more clarity on when/what we're going to offload and how this relates to consumer schemas (contracts) to avoid breaking consumers as much as possible.